### DISCUSSION TITLE
请问自己制作的第三方 xmake 包内部支持了 releasedbg，add_requires 的时候如何配置 releasedbg 模式

### DISCUSSION THREAD

**Discussion by ivanallen** [xmake-io/xmake#4041] [Q&A]
请问自己制作的第三方 xmake 包内部支持了 releasedbg，add_requires 的时候如何配置 releasedbg 模式？

使用 `xmake l scripts/test.lua --shallow -vD -m releasedbg xxx` 进行测试的时候，发现编译的还是 release 模式

![image](https://github.com/xmake-io/xmake/assets/12481610/a748fa29-88ef-4a4a-9ca9-5dd025652056)

编译过程中依然是 release

![image](https://github.com/xmake-io/xmake/assets/12481610/f5a5efcd-9496-4b79-917b-e8eaac2799ca)

**Comment by star-hengxing** [comment]
https://github.com/xmake-io/xmake/blob/0ffb589fb816c4d886cf88d322df45f7ebe5434e/xmake/modules/package/tools/xmake.lua#L203-L205

观察源码，需要自己通过 configs 传进去：

```lua
local configs = {}
configs.mode = "$(mode)"
import("package.tools.xmake").install(package, configs)
```
